addons/medical_vitals/functions/fnc_updateHeartRate.sqf

```diff
@@ -57,23 +58,26 @@ if !IN_CRDC_ARRST(_unit) then {
             _targetBP = _targetBP * (_bloodVolume / DEFAULT_BLOOD_VOLUME);
         };
 
-        private _targetHR = DEFAULT_HEART_RATE;
+        _targetHR = DEFAULT_HEART_RATE;
         if (_bloodVolume < BLOOD_VOLUME_CLASS_2_HEMORRHAGE) then {
             _targetHR = _heartRate * (_targetBP / (45 max _meanBP));
         };
         if (_painLevel > 0.2) then {
             _targetHR = _targetHR max (80 + 50 * _painLevel);
         };
-        _targetHR = _targetHR + _hrTargetAdjustment;
+        _targetHR = (_targetHR + _hrTargetAdjustment) max 0;
 
         _hrChange = round(_targetHR - _heartRate) / 2;
     } else {
         _hrChange = -round(_heartRate / 10);
     };
-    _heartRate = (_heartRate + _deltaT * _hrChange) max 0;
+    if (_hrChange < 0) then {
```

Comment:
You can put the whole new If statement directly under the _hrChange calculation and spare it since it is redundant:
```
   _hrChange = round(_targetHR - _heartRate) / 2;
   _heartRate = (_heartRate + _deltaT * _hrChange) min _targetHR;
} else ...

Reply:
I'm sorry, I don't think I understand what you are suggesting. The heart rate is currently doing a min or max with _targetHR depending on if we are increasing or decreasing our heart rate. Wouldn't the change you are suggesting here cause any gradual change lowering the HR to happen immediately instead?

Reply:
After going through the code more closely these lines look good to me 👍

This doesn't fix the issue of incorrectly large `deltaT` first tick after cardiac arrest, but you're right that we don't want the heart rate to pass the target anyway so it should stay because it does fix that issue (theoretically `deltaT` could reach 10 at any time).

For the exiting cardiac arrest `deltaT` issue, I think we should still be running `handleUnitVitals` while in cardiac arrest because it appears to handle other things not related to heart rate too, if we don't then we should refresh the stored `lastTimeUpdated` value on the unit when exiting cardiac arrest to avoid similar issues to the one revealed here.